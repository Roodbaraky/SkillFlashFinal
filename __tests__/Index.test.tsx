import Index from "../app/index";
import { fireEvent, render, screen } from "@testing-library/react-native";
import React from "react";
import "@testing-library/react-native/extend-expect";

test("Index it has two buttons", async () => {
  render(<Index />);
  expect(screen.getByText("Sign up")).toBeOnTheScreen();
  expect(screen.getByText("Log in")).toBeOnTheScreen();
});

test("opens login component", async () => {
  render(<Index />);
  fireEvent.press(screen.getByText("Log in"));
  expect(screen.getByTestId("login-container")).toBeOnTheScreen();
});

test("opens sigup component", async () => {
  render(<Index />);
  fireEvent.press(screen.getByText("Sign up"));
  expect(screen.getByTestId("signup-container")).toBeOnTheScreen();
});
