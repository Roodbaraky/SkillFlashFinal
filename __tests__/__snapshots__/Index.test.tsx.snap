// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignSelf": "center",
      "flex": 1,
      "paddingTop": 58,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "black",
        "fontSize": 50,
        "marginBottom": 20,
        "marginTop": 100,
      }
    }
  >
    SkillFlash
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "borderStyle": "solid",
          "borderWidth": 5,
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Log in
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "borderStyle": "solid",
          "borderWidth": 5,
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Sign up
    </Text>
  </View>
</RNCSafeAreaView>
`;
